test/plain.jl

```diff
@@ -332,4 +332,12 @@ using Compat.String
         str = read(fid["test"])
         @test str == "Hello World"
     end
+    
+    # Test Float16 support
+    arr_float16 = Float16[1.0, 0.25, 0.5, 8.0]
+    h5write("test_float16.h5", "x", arr_float16)
```

Comment:
should clean this file up when the test is done, otherwise stale results from previous runs could give misleading results